File: QE-Python_selenium/qe-selenium-python/Activity10.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains

driver = webdriver.Firefox()
driver.get("https://training-support.net/webelements/drag-drop")

print("Page Title:", driver.title)

actions = ActionChains(driver)

# Locate elements
ball = driver.find_element(By.ID, "ball")
d1 = driver.find_element(By.ID, "dropzone1")
d2 = driver.find_element(By.ID, "dropzone2")

# Drag and drop into Dropzone 1
actions.drag_and_drop(ball, d1).perform()

# Verify ball is inside Dropzone 1
if len(d1.find_elements(By.ID, "ball")) > 0:
    print("Ball dropped into Dropzone 1")
else:
    print("Ball NOT dropped into Dropzone 1")

# Re-locate ball (DOM can change after drag-drop)
ball = driver.find_element(By.ID, "ball")

# Drag and drop into Dropzone 2
ActionChains(driver).drag_and_drop(ball, d2).perform()

# Verify ball is inside Dropzone 2
if len(d2.find_elements(By.ID, "ball")) > 0:
    print("Ball dropped into Dropzone 2")
else:
    print("Ball NOT dropped into Dropzone 2")

driver.quit()

File: QE-Java-Selenium/QE-selenium-java/javaselenium/src/main/java/javaselenium/demos/Activity10.java
package javaselenium.demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public class Activity10 {
    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        driver.get("https://training-support.net/webelements/drag-drop");

        System.out.println("Page Title: " + driver.getTitle());

        Actions builder = new Actions(driver);

        WebElement ball = driver.findElement(By.id("ball"));
        WebElement d1 = driver.findElement(By.id("dropzone1"));
        WebElement d2 = driver.findElement(By.id("dropzone2")); 

        builder.dragAndDrop(ball, d1).perform(); 
        if (d1.findElements(By.id("ball")).size() > 0) {
            System.out.println("Ball dropped into Dropzone 1");
        } else {
            System.out.println("Ball NOT dropped into Dropzone 1");
        }
 
        ball = driver.findElement(By.id("ball"));
        builder.dragAndDrop(ball, d2).perform();
        if (d2.findElements(By.id("ball")).size() > 0) {
            System.out.println("Ball dropped into Dropzone 2");
        } else {
            System.out.println("Ball NOT dropped into Dropzone 2");
        }

        driver.quit();
    }
}